[ET-VK][Ops] dequantize_per_channel reference impl and testing

ahmtox opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #12369
* #12210
* #12209
* #12208
* #12207
* __->__ #12206
* #12205
* #12204
* #12203
* #12201
* #12200
* #12199

# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

Commit:
[ET-VK][Ops] dequantize_per_channel reference impl and testing

# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

lucylq reviewed (approved):
LGTM on the executorch/extension side, thank you for extending the functionality to include const ref optionals and adding tests coverage.

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_channel reference impl and testing"


# Context

In order to properly enable dynamic quantization, we create the dequantize_per_channel operator as its seemingly useful to have for the pipeline.

# Changes

This creates the wrapper for the cpu reference implementation, and also a dummy reference implementation I created just to test against it.

Differential Revision: [D77746138](https://our.internmc.facebook.com/intern/diff/D77746138/)

[ghstack-poisoned]
